data/modules/SecondHand/SecondHand.lua

```diff
@@ -44,29 +43,54 @@ local onDelete = function(ref)
 	ads[ref] = nil
 end
 
--- check it fits on the ship, both the slot for that equipment and the
--- weight.
+-- Function: canFit
```

Comment:
Note that this style of documentation comment is unnecessary as this is an internal function of a specific module. The NaturalDocs style is used to document public API functions and documentation using its keywords winds up in the [codedoc](https://codedoc.pioneerspacesim.net) - this function is completely inaccessible outside of this Lua module and thus may actually generate misleading documentation for this function representing it as an API function.

Reply:
An automated documentation system should be able to distinguish between public and private functions and the output should be configurable to output just the public API or all APIs. At least that's how `Doxygen` works.

Apart from the rather obvious "local" declaration of the function, is there a way to tell codedoc to not publish a documentation block to the API docs?

I'm a firm believer of good documentation as it helps people (like me) who have a poor grasp of a language.

----

TL;DR: having just read some of the Natural Docs (https://www.naturaldocs.org/) documentation, I guess just removing the "Function:" line will fix this?

Reply:
Yep - NaturalDocs is intentionally not smart enough to extract semantic meaning from the source file; it merely regexes the function signature you're documenting.

That fix should work; I was mentioning it more to say that you don't need to go through the formalism of reproducing the codedoc structure since the comment will only ever be useful to someone looking at the code directly - and thus the focus should be on clearly communicating the *intent* of the function rather than repeating obvious facts (like the parameter list) for the benefit of an audience that doesn't/can't read the implementation.